ProjFS.Mac/PrjFSKextLogDaemon/PrjFSKextLogDaemon.cpp

```diff
@@ -221,12 +239,141 @@ static bool TryFetchAndLogKextHealthData(io_connect_t connection)
+        "}");
+        
     }
     else
     {
+        WriteJsonToMessageListener(
+        "{"
+            "\"ErrorMessage\":\"Fetching profiling data from kernel failed\","
+            "\"ret\":\"" + to_string(ret) + "\","
+        "}");
+        
         fprintf(stderr, "fetching profiling data from kernel failed: 0x%x\n", ret);
         return false;
     }
     
     return true;
 }
+
+static void CreatePipeToMessageListener()
+{
+    if (INVALID_SOCKET_FD != s_messageListenerSocket)
+    {
+        // Already connected
+        return;
+    }
+
+    s_messageListenerSocket = socket(PF_UNIX, SOCK_STREAM, 0);
+    if (s_messageListenerSocket < 0)
+    {
+        os_log_with_type(
+            s_kextLogger,
+            OS_LOG_TYPE_DEFAULT,
+            "Failed to create a new socket, path: %s, error: %d",
+            s_messageListenerSocketPath.c_str(),
+            errno);
+        
+        s_messageListenerSocket = INVALID_SOCKET_FD;
+        return;
+    }
+    
+    struct sockaddr_un socket_address;
+    memset(&socket_address, 0, sizeof(struct sockaddr_un));
+    
+    socket_address.sun_family = AF_UNIX;
+    size_t resultLength = strlcpy(socket_address.sun_path, s_messageListenerSocketPath.c_str(), sizeof(socket_address.sun_path));
+    
+    if (resultLength >= sizeof(socket_address.sun_path))
+    {
+        os_log_with_type(
+            s_kextLogger,
+            OS_LOG_TYPE_DEFAULT,
+            "Could not copy socket path: %s, insufficient buffer. resultLength: %lu, sizeof(socket_address.sun_path): %lu",
+            s_messageListenerSocketPath.c_str(),
+            resultLength,
+            sizeof(socket_address.sun_path));
+        
+        goto ClosePipeAndCleanup;
+    }
+    
+    if(0 == connect(s_messageListenerSocket, (struct sockaddr *) &socket_address, sizeof(struct sockaddr_un)))
+    {
+        os_log_with_type(
+            s_kextLogger,
+            OS_LOG_TYPE_DEFAULT,
+            "Connected to message listener on socket '%s'",
+            s_messageListenerSocketPath.c_str());
+        return;
+    }
+    
+    os_log_with_type(
+        s_kextLogger,
+        OS_LOG_TYPE_DEFAULT,
+        "Failed to connect socket, pipeName: %s, error: %d",
+        s_messageListenerSocketPath.c_str(),
+        errno);
+    
+ClosePipeAndCleanup:
+
+    if (INVALID_SOCKET_FD != s_messageListenerSocket)
+    {
+        close(s_messageListenerSocket);
+        s_messageListenerSocket = INVALID_SOCKET_FD;
+    }
+}
+
+static void WriteJsonToMessageListener(const string& jsonMessage)
+{
+    if (INVALID_SOCKET_FD == s_messageListenerSocket)
+    {
+        return;
+    }
+
+    // TODO: Properly version PrjFSKextLogDaemon and other Mac binaries
+    string fullMessage =
+    "{\""
+        "version\":\"0.6.XXX.X\","
+        "\"providerName\":\"Microsoft.Git.GVFS\","
+        "\"eventName\":\"PrjFSKextLogDaemon\","
+        "\"eventLevel\":2,"
```

Comment:
You might check, but IIRC I got ride of the event level and opcode fields.

Reply:
(On my iPad right now and can’t check...)

Reply:
From what I can tell these are gone, but when you're back in the office could you confirm I read the code correctly? 😄

Reply:
> You might check, but IIRC I got ride of the event level and opcode fields.

@jeffhostetler if my manual testing looks good I will assume that I read the service daemon code correctly and don't need to set these fields.  If it ends up that I do still need to set these please let me know 😃

Reply:
yeah, the event level and opcode are gone.